File: iotOnGolang/internal/model/NetworkProfile.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// NetworkProfile Declaration
//==============================================================
type NetworkProfile struct {
    gorm.Model
     ProfileName                                    string
    Ssid                                    string
    Apn                                    string
    DeviceId         *uint
    Device           *IoTDevice `gorm:"foreignKey:DeviceId"`
    GatewayId         *uint
    Gateway           *Gateway `gorm:"foreignKey:GatewayId"`
    SimCardId         *uint
    SimCard           *SimCard `gorm:"foreignKey:SimCardId"`
    ConnectivityType                      ConnectivityType

// parent associations as their child

}

File: src/app/models/NetworkProfile.js


// Define collection and schema for NetworkProfile
export interface NetworkProfile {
    profileName:
	type : string
    ssid:
	type : string
    apn:
	type : string
    Device:
	type : Schema.Types.ObjectId
    Gateway:
	type : Schema.Types.ObjectId
    SimCard:
	type : Schema.Types.ObjectId
    ConnectivityType:
 	type : String
#
    collection: 'networkProfiles'
}
